activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,10 @@
+*   Warn about changing `query_constraints:` behavior
```

Comment:
This should explain the change rather than the behavior. 

```suggestion
*   In association option `query_constraints` is deprecated in favor of `foreign_key`.
```